lldb-eval/eval_test.cc

```diff
@@ -273,6 +273,56 @@ Matcher<EvalResult> IsError(std::string value) {
   return MakeMatcher(new IsErrorMatcher(std::move(value)));
 }
 
+class IsTypeMatcher : public MatcherInterface<EvalResult> {
```

Comment:
Maybe we can just enable type comparison with lldb-eval and LLDB already? Are there cases where the resulting type is different and we can't do anything about it?